konrad/cloud.py

```diff
@@ -90,6 +104,7 @@ def __init__(self, z, cloud_fraction=0,
             single_scattering_albedo (float / DataArray): single scattering
                 albedo due to cloud (for the shortwave component of RRTMG)
         """
+        self.z = z  # TODO: Can we circumvent saving `z`?
         numlevels = z.size
```

Comment:
I remember discussing this with you before, because you didn't want the user to have to input the number of model levels. Something is needed so that we can set defaults for properties that are irrelevant for the type of cloud being created. The reason I used z was for preserving the cloud mass, given physical cloud parameters. If we decide to use and preserve cloud optical properties instead, this isn't necessary. How do you feel about using p instead? Alternatively, we could get the size from the cloud fraction array (needed for both "physical" and "direct input" clouds), forcing the user to input an array for cloud fraction (which seems a bit unnecessary for clear sky simulations).

Reply:
Sorry for blaming you for my decision! But I still think that past-me was wrong; or at least that the current implementation is not the nicest.

We could use an `Atmosphere` to initialise the clouds. There will never be a run without an atmosphere and we are already using it to update the cloud profile.

Reply:
Yeah, I'm also fine with doing that :)